Mathlib/NumberTheory/SmoothNumbers.lean

```diff
@@ -265,6 +265,23 @@ lemma mem_smoothNumbers_iff_forall_le {n m : ℕ} :
 lemma mem_smoothNumbers' {n m : ℕ} : m ∈ smoothNumbers n ↔ ∀ p, p.Prime → p ∣ m → p < n := by
   simp only [smoothNumbers_eq_factoredNumbers, mem_factoredNumbers', Finset.mem_range]
 
+/-- The prime factors of `m > 1` are contained in the set of primes below `n`
+  if `m` is a `n`-smooth number -/
+lemma primeFactors_subset_of_mem_smoothNumbers (m n : ℕ) (hms : m ∈ n.smoothNumbers) :
+    m.primeFactors ⊆ n.primesBelow :=
+  have hxle (m n x : ℕ) (hpf : x ∈ m.primeFactors) (hms : m ∈ smoothNumbers n) :
+      x < n :=
+    match mem_smoothNumbers.mp hms with
+    | ⟨ _, h2 ⟩  => h2 x (mem_primeFactors_iff_mem_factors.mp hpf)
+  fun w hh => mem_primesBelow.mpr ⟨hxle m n w hh hms, prime_of_mem_primeFactors hh⟩
```

Comment:
```suggestion
lemma primeFactors_subset_of_mem_smoothNumbers {m n : ℕ} (hms : m ∈ n.smoothNumbers) :
    m.primeFactors ⊆ n.primesBelow :=
  have hxle {x : ℕ} (hpf : x ∈ m.primeFactors) (hms : m ∈ n.smoothNumbers) :
      x < n :=
    match mem_smoothNumbers.mp hms with
    | ⟨_, h⟩  => h x (mem_primeFactors_iff_mem_factors.mp hpf)
  fun _ h => mem_primesBelow.mpr ⟨hxle h hms, prime_of_mem_primeFactors h⟩
```
Notes:
- Use implicit variables when it makes sense (by using `{` and `}`  instead of `(` and `)`).
- Why use `h'` or `h2` for hypothesis names when `h` is available?